utils/utils.go

```diff
@@ -0,0 +1,397 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Each resource request will be fetched from the Kubernetes API Server at a
+// max of 20 retries, with 15 seconds in between each attempt.
+// This creates a max wait time of upto 5 minutes that a resource request
+// must successfully return within, before moving on.
+
+// MaxRetries is the maximum number of retries that a resource will be
+// attempted to be fetched from the Kubernetes API Server.
+const MaxRetries = 20
+
+// RetryInterval is the number of seconds to sleep in between requests
+// to the Kubernetes API Server.
+const RetryInterval = 15
+
+// RunEKSSmokeTest instantiates the EKS Smoke Test.
+func RunEKSSmokeTest(t *testing.T, resources []apitype.ResourceV3, kubeconfigs ...interface{}) {
```

Comment:
Somewhat of a nit - but seems like nearly everything here should be private?  Would make clearer what the api of this module is.

Reply:
Great points - the idea was to make this as generic and modular as possible, to be able to leverage the validation code here for other managed k8s providers in the future, hence the many public / exported methods.

I'll clean this up and only expose the proper methods suitable to this pkg API

Reply:
Done